### ISSUE TITLE
dereference_dn and RecursionError

### ISSUE THREAD

**Issue by daenney** [cannatag/ldap3#419]
I'm trying to use `dereference_dn` to, on a query for a user (an `inetOrgPerson`), have it derefence the `manager` attribute which points to another `inetOrgPerson`. I could also manually do that but I'd rather the LDAP server take care of it instead of me issuing a second query.

The `ObjectDef` looks like this:

```
object_class: ['inetOrgPerson', 'posixAccount', 'shadowAccount']
    AttrDef(key='uid', name='uid'),
    AttrDef(key='name', name='cn'),
    AttrDef(key='location', name='l'),
    AttrDef(key='email', name='mail'),
    AttrDef(key='manager', name='manager', dereference_dn=object_class: ['inetOrgPerson', 'posixAccount', 'shadowAccount'])
``` 

The thing I pass to the `AttrDef` as `dereference_dn` is another `ObjectDef`:

```
manager_ = ObjectDef([
        'inetOrgPerson',
        'posixAccount',
        'shadowAccount'
    ])
```

So the `manager` field is `AttrDef(..., dereference_dn=manager_)`.

However, when I now do a query I end up with `RecursionError: maximum recursion depth exceeded while calling a Python object`

But I don't understand why. Any DN that gets pointed at by the `manager` field can in itself have a `manager` defined on it, but since I'm not asking for that to be dereferenced in the `manager_` `ObjectDef`, how does it end up in a recursive loop like this?

**Comment by cannatag** [comment]
Hi, probably you have a inetOrgPerson object that has itself as a manager, so when dereferencing the manager (in the first ObjectDef) you keep requesting the same object until Python raises the maximum recursion depth exception.

**Comment by cannatag** [comment]
I've added a check to avoid the maximum recursion depth. Now if an object is referencing itself an LDAPObjectReferenceError is raised. 

The fix will be included in the upcoming version 2.4 of ldap3.

Thanks,
Giovanni

**Comment by cannatag** [comment]
Fixed in version 2.4
